### ISSUE TITLE
Patch IDs on irregular grids

### ISSUE THREAD

**Issue by SiccarPoint** [landlab/landlab#308]
The data structures for patches on irregular grids are very broken. e.g., 
```
>>> mg = HexModelGrid(3,3)  # bottom row 3, mid row 4, top row 3 nodes
>>> mg.nodes_at_patch[0, :]
array([4, 7, 3])  # think this should be [4, 3, 0]
>>> mg.patches_at_node[4, :]
array([0, 1, 2, 3, 8, 9])  # think this should be [7, 6, 5, 0, 1, 2]
```

**Comment by SiccarPoint** [comment]
This is being addressed in SiccarPoint/modernise-components.

**Comment by SiccarPoint** [comment]
Fixed in #314
